### PR TITLE
fix: form spacing

### LINKED ISSUE #2343: Form spacing inconsistencies
## Detailed description

> Describe in detail the issue you're having.

Fieldsets and form items should have consistent spacing within a form.  Currently Fieldsets have a margin-bottom of `$carbon--spacing-07` (https://github.com/IBM/carbon-components/blob/master/src/components/form/_form.scss#L20) but form-item doesn't have any margin-bottom meaning we have to manually add it.  The demos on the Carbon site have a manual 2rem `margin-bottom` added to form-item.

Secondly according to the guidelines, the spacing should be `24px/1.5rem` which would be `$carbon--spacing-06`.

> Is this issue related to a specific component?

Form

> What did you expect to happen? What happened instead? What would you like to see changed?

Form items and fieldsets have consistent spacing (either they both have it or neither have it) and follow the recommended guidelines outlined on the Carbon site. https://www.carbondesignsystem.com/components/form/style#recommended

> What browser are you working in?

Chrome

> What version of the Carbon Design System are you using?

v10.1.0

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?

Watson Health Design Pattern & Asset Library (PAL)

## Steps to reproduce the issue

1. Inspect the form demo on the carbon site.
1. note the margin bottom is not part of the component styles
1. note the margin bottom value does not align with the recommended values

### PR DESCRIPTION
**Pull request by breid1313**
Closes #2343 

#### Changelog

**Changed**

- added 24px/1.5 rem `margin-bottom` to `form--form-item` using the `$carbon--spacing-06` following the [Form Style Guide](https://www.carbondesignsystem.com/components/form/style/#recommended)
- adjusted the `margin-bottom` on `form--fieldset` to `$carbon--spacing-06` for consistency

#### Testing / Reviewing

Storybook before (small snippet):
Screen Shot 2019-11-05 at 3 57 01 PM

Storybook after (small snippet):
Screen Shot 2019-11-05 at 3 56 10 PM

### CODE REVIEW

File: packages/components/src/components/form/_form.scss [hunk 1]
```diff
@@ -29,6 +29,7 @@
     // `flex-direction: row`
     flex: 1 1 auto;
     align-items: flex-start;
+    margin-bottom: $carbon--spacing-06;
```
Comment (joshblack) [inline packages/components/src/components/form/_form.scss:32]: Would adding this to `bx--form-item` unexpected add margin for folks using this control? Might be good to offer it by default, but might catch folks off guard if we add this layout potentially

**Review by aagonzales — APPROVED** [review]
If dev thinks its ok to update without breaking people then it good to me.

File: packages/components/src/components/form/_form.scss (hunk 1 above)
Comment (emyarod) [reply packages/components/src/components/form/_form.scss:32]: yeah I think this is part of a discussion that's been on hold since the v10 effort began

**Comment by joshblack** [comment]
Coming back to this, looks like we can't update it without causing a breaking change for certain users 😞Went ahead and added a `v11` label to the corresponding issue so we make sure to address this in our next update.

Just wanted to say thanks so much for your time and effort here, really appreciate it 🙏 Sorry that things took so long and ended up the way that they did 😞

**Comment by breid1313** [comment]
@joshblack no need to apologize! Happy to (continue to) help where I can 👍 . Thanks for following up 😄

**Pull request closed without merging**